TombEngine/Scripting/Internal/TEN/Misc/Miscellaneous.cpp

```diff
@@ -358,6 +361,37 @@ namespace Misc
 		TENLog(message, level, LogConfig::All, USE_IF_HAVE(bool, allowSpam, false));
 	}
 
+	static void DrawSprite(GAME_OBJECT_ID objNum, unsigned int spriteIndex, const Vec2& pos, float orient2D, ScriptColor color, const Vec2& size)
+	{
+		TENLog("--Test DrawSprite--", LogLevel::Info);
+		TENLog("GAME_OBJECT_ID: " + objNum, LogLevel::Info);
+		TENLog("spriteIndex: " + std::to_string(spriteIndex), LogLevel::Info);
+		TENLog("Pos x " + std::to_string(pos.x), LogLevel::Info);
+		TENLog("Pos y " + std::to_string(pos.y), LogLevel::Info);
+		TENLog("orient2D " + std::to_string(orient2D), LogLevel::Info);
+		TENLog("Color:", LogLevel::Info);
+		TENLog("R: " + std::to_string(color.GetR()), LogLevel::Info);
+		TENLog("G: " + std::to_string(color.GetG()), LogLevel::Info);
+		TENLog("B: " + std::to_string(color.GetB()), LogLevel::Info);
+		TENLog("A: " + std::to_string(color.GetA()), LogLevel::Info);
+		TENLog("Size x " + std::to_string(size.x), LogLevel::Info);
+		TENLog("Size y " + std::to_string(size.y), LogLevel::Info);
+		TENLog("--Test DrawSprite--", LogLevel::Info);
+		auto orient2 = ANGLE(orient2D);
+		auto color2 = Vector4(color.GetR(), color.GetG() , color.GetB(), color.GetA());
+		g_Renderer.DrawSpriteIn2DSpace(objNum, spriteIndex, Vector2(pos.x,pos.y), orient2, color2, Vector2(size.x,size.y));
```

Comment:
You can't directly call draw functions from a script or any part of the game logic loop (`ControlPhase`). Sprite drawing must be queued (sprites should be collected in a temporary list) and rendered when `RenderScene` is called.